api_core/tests/unit/test_operation.py

```diff
@@ -113,6 +116,23 @@ def test_result():
     assert future.done()
 
 
+def test_done_w_retry():
+    RETRY_PREDICATE = retry.if_exception_type(exceptions.TooManyRequests)
+    test_retry = retry.Retry(predicate=RETRY_PREDICATE)
```

Comment:
Instancing custom retry to avoid collision with default value and to make check clear